hnn_core/gui/gui.py

```diff
@@ -26,15 +26,307 @@
+                                                 height='30px',
+                                                 button_color=self.theme_color)
+
+        self.load_button = FileUpload(accept='.json,.param',
+                                      multiple=False,
+                                      style={'button_color': self.theme_color},
+                                      description='Load network',
+                                      button_style='success')
+
+        self.delete_drive_button = create_expanded_button(
+            'Delete drives', 'success', height='30px',
+            button_color=self.theme_color)
+
+        # Visualization figure list
+        self.plot_outputs_list = list()
+        self.plot_dropdowns_list = list()
+        # Add drive section
+        self.drive_widgets = list()
+        self.drive_boxes = list()
+
+        self._init_ui_components()
+
+    def _init_ui_components(self):
+        """Initialize larger UI components and dynamical output windows. It's
+        not encouraged for users to modify or access attributes in this part.
+        """
+        self.connectivity_sliders = self.init_cell_connectivity(self.params)
+        # dynamic larger components
+        self.drives_out = Output()  # window to add new drives
+        self.log_out = Output(
+            layout={
+                'border': '1px solid gray',
+                'height': self.log_window_height,
+                'overflow': 'auto'
+            })
+        # visualization window
+        self.visualization_window = Output(layout={
+            'height': self.visualization_window_height,
+            'width': self.visualization_window_width,
+        })
+        # detailed configuration of backends
+        self.backend_config = Output()
+
+        # static parts
+        # Running status
+        self.simulation_status = HTML(value="""<div
+            style='background:gray;padding-left:10px;color:white;'>
+            Not running</div>""")
+
+        # footer
+        self.footer = VBox([
+            HBox([
+                HBox([
+                    self.run_button, self.load_button, self.delete_drive_button
+                ], layout={"width": self.left_sidebar_width}),
+                self.viz_layout_selection,
+            ]), self.log_out, self.simulation_status
+        ])
+        # title
+        self.header = HTML(value=f"""<div
+            style='background:{self.theme_color};
+            text-align:center;color:white;'>
+            HUMAN NEOCORTICAL NEUROSOLVER</div>""")
+
+    def load_parameters(self, params_fname=None):
+        if not params_fname:
+            # by default load default.json
+            hnn_core_root = op.join(op.dirname(hnn_core.__file__))
+            params_fname = op.join(hnn_core_root, 'param', 'default.json')
+
+        return read_params(params_fname)
+
+    @staticmethod
+    def init_cell_connectivity(params):
+        return [
+            _get_sliders(params, [
```

Comment:
Relevant to https://github.com/jonescompneurolab/hnn-core/issues/497#issuecomment-1186600445